### DISCUSSION TITLE
Having Trouble Getting WebRTC to work

### DISCUSSION THREAD

**Discussion by stawze** [bluenviron/mediamtx#3438] [Questions]
### Question

So I'm running mediamtx in a docker container. I've opened ports 8889/tcp and 8189/udp on my firewall and they pass a connection test
When I run mediamtx and try to publish to http://_server-ip_:8889/stream/whip I got the following error:
```
2024/06/08 23:00:12 INF [WebRTC] [session 63377dc5] created by _my-public-ip_:50455
2024/06/08 23:00:22 INF [WebRTC] [session 63377dc5] closed: deadline exceeded while waiting connection
```
I looked around and found the following: https://github.com/bluenviron/mediamtx#connectivity-issues
So I created a mediamtx.yml file and added the line:
`webrtcAdditionalHosts: [_server-ip_]`
However now when I run the container and pass the file in and try and publish to the server I get this error:
```
2024/06/08 23:12:45 INF [WebRTC] [session 29a04e02] created by _my-public-ip_:50528
2024/06/08 23:12:45 INF [WebRTC] [session 29a04e02] closed: path 'stream' is not configured
```
I looked around in the documentation, issues and discussion page again but couldn't find anything
Is anyone able to lend me a hand here?
Docker command I'm running is: `docker run --rm -it --network=host -v $PWD/mediamtx.yml:/mediamtx.yml bluenviron/mediamtx:latest`
Thanks.

**Comment by stawze** [comment]
I added the extra two lines to my config and now it works:
```
paths:
  stream:
```
However the stream is extremely laggy and constantly drops frames, even after experimenting with the encoding settings
![image](https://github.com/bluenviron/mediamtx/assets/90922963/89259476-73d3-4a76-9716-7cb9de9133d6)

**Reply by aler9** [reply]
Hello, lags and drops may be caused by
* the inability of your H264 encoder to keep up with the speed of the stream. The zerolatency tune is known for dropping frames when the encoder is too slow, preserving real-time (that's why it's called "zerolatency"). Try removing zerolatency or switching from software encoder (x264) to hardware encoder (check in the video encoder dropdown menu)
* insufficient bandwidth between OBS and MediaMTX, or between MediaMTX and the readers. It's impossible to know for sure without additional data. Decrease the Bitrate in order to find out whether this is the issue.

[a comment by an automated account is not shown]
